### ISSUE TITLE
[example] First message does not reach friends not being the bootstrapper

### ISSUE THREAD

**Issue by najeal** [commonwarexyz/monorepo#45]
## What happens

Testing the chat example:
when a new friend joins the network, his first message only reaches the bootstrapper used by the command.

## How to reproduce

```
cargo run --release -- --me=1@3001 --friends=1,2,3,4

-> i am 1

cargo run --release -- --me=2@3002 --friends=1,2,3,4 --bootstrappers=1@127.0.0.1:3001

-> i am 2

cargo run --release -- --me=3@3003 --friends=1,2,3,4 --bootstrappers=1@127.0.0.1:3001

-> i am 3
-> i am again 3
```

`i am 3` will only be displayed in the Friend-1 terminal
`I am again 3` will be displayed in both Friend-1 and Friend-2 terminal

**Comment by patrick-ogrady** [comment]
Ah, I probably need to improve the docs here (and might be a good way for you to contribute lol).

The `commonware-chat` is very basic right now. It only relays messages to peers that are currently connected (with no replay). It would be cool if messages were stored client-side and peers could request previous messages when connecting.

**Comment by patrick-ogrady** [comment]
I should really call it an "ephemeral chat app," but that didn't sound catchy enough.

**Comment by najeal** [comment]
I would be glad to try to add this local storage !

But in the issue I have seen: friend-2 does not receive the first message from friend-3 whereas he has connected before friend-3.

**Comment by patrick-ogrady** [comment]
> But in the issue I have seen: friend-2 does not receive the first message from friend-3 whereas he has connected before friend-3.

It is still possible for friend-2 not to be connected to friend-3 by the time friend-3 sends a message (based on aggressiveness of config/random sleep). As a sanity check, make sure to check the `p2p_connections` metric at the top right of the application (wait for all of them to say `p2p_connections 2` before sending the first message).  I tried to replicate your issue once everyone was connected and could not.

> I would be glad to try to add this local storage !

A simpler and more useful alternative to what I suggested earlier, would be to add some sort of backlog where messages are pooled on the sender side for a recipient until it connects. Upon connection, all messages can be sent.

******
Also recommend checking out the new docs on how p2p works: https://docs.rs/commonware-p2p/latest/commonware_p2p/#design
